.github/workflows/build.yml

```diff
@@ -65,6 +65,30 @@ jobs:
           name: metadata.json
           path: ${{ steps.generate-metadata-file.outputs.filepath }}
 
+  # This job uploads an Actions artifact named terraform-registry-manifest.json.zip.
+  # The artifact contains a single file with a filename that Terraform Registry expects
+  # (example: terraform-provider-cloudinit_2.3.6-alpha1.json) and contents identical
+  # to the terraform-registry-manifest.json file in the repository.
+  upload-terraform-registry-manifest-file:
+    needs: set-product-version
+    runs-on: ubuntu-latest
+    outputs:
+      filepath: ${{ steps.terraform-registry-manifest.outputs.filename }}
+    steps:
```

Comment:
After we build confidence that this works, we can extract it to a reusable workflow. No need to copy/paste/maintain everywhere 😃